packages/block-library/src/utils/use-template-context.js

```diff
@@ -0,0 +1,325 @@
+/**
+ * WordPress dependencies
+ */
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+
+const DEFAULT_CONTEXT = {
+	taxonomy: null,
+	termSlug: null,
+	isAuthor: false,
+	authorSlug: null,
+};
+
+/**
+ * Creates a template context object by merging provided context with default context.
+ *
+ * @param {Object} providedContext Provided context object to merge with default context.
+ * @return {Object} Complete context object with all required fields.
+ */
+function buildTemplateContext( providedContext = {} ) {
+	return {
+		...DEFAULT_CONTEXT,
+		...providedContext,
+	};
+}
+
+/**
+ * Hook to get the current template slug from the editor context.
+ *
+ * @return {string|null} The current template slug or null if not available.
+ */
+export function useTemplateSlug() {
```

Comment:
A small optimization would be to pass `templateSlug` from block context, if available and avoid the queries in that case.

Reply:
>We'd need to add templateSlug to the usesContext attribute for each of these blocks. 

That's what I had in mind, yes. It should be fine since they need this.

Reply:
Thanks! In https://github.com/WordPress/gutenberg/pull/72549/commits/b6b6d31885dd11e1488c4e7a1eae5bf926d35235, I've passed `templateSlug` from the block context of these blocks so it can be used in `useTemplateSlug ` if available.